test/verify/check-networkmanager-wireguard

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3 -cimport os, sys; os.execv(os.path.dirname(sys.argv[1]) + "/../common/pywrap", sys.argv)
+
+from testlib import MachineCase, test_main
+
+
+class TestWireGuard(MachineCase):
+    provision = {
+        "machine1": {"address": "192.168.100.11/24", "restrict": False},
+        "machine2": {"address": "192.168.100.12/24", "restrict": False},
+    }
+
+    def testVPN(self):
+        m1 = self.machines["machine1"]
+        m2 = self.machines["machine2"]
+        b = self.browser
+
+        self.login_and_go("/network")
+
+        # Peer 2 (server)
+        m2.execute("systemctl stop firewalld") # TODO: Fix it
+        m2.execute("wg genkey > private")
+        m2_port = 51820
+        m2_pubkey = m2.execute("wg pubkey < private").strip()
+
+        m2.execute("ip link add dev wg0 type wireguard")
+        m2.execute("ip addr add 10.0.0.2/24 dev wg0")
+        m2.execute("wg set wg0 private-key ./private")
+        m2.execute(f"wg set wg0 listen-port {m2_port}")
+        m2.execute("ip link set wg0 up")
+
+        # Peer 1 (client)
+        m1.execute("systemctl stop firewalld") # TODO: Fix it
+        b.click("button:contains('Add WireGuard')")
+        b.wait_visible("#network-wireguard-settings-dialog")
+        b.click("button:contains('Generate private key')")
+        b.wait_not_val("#network-wireguard-settings-dialog .pf-v5-c-clipboard-copy input", "")
+        m1_pubkey = b.eval_js("document.querySelector('#network-wireguard-settings-dialog .pf-v5-c-clipboard-copy input').value")
+        b.set_input_text("#network-wireguard-settings-listen-port-input", "51820")
+        b.set_input_text("#network-wireguard-settings-addresses-input", "10.0.0.1/24")
+        b.click("button:contains('Add peer')")
+        b.set_input_text("#network-wireguard-settings-publickey-peer-0", m2_pubkey)
+        b.set_input_text("#network-wireguard-settings-endpoint-peer-0", f"192.168.100.12:{m2_port}")
+        b.set_input_text("#network-wireguard-settings-allowedips-peer-0", "10.0.0.2")
+        b.click("button:contains('Save')")
+
+        m2.execute(f"wg set wg0 peer {m1_pubkey} allowed-ips 10.0.0.1/32") # endpoint and port is not necessary for a peer if that peer estalishes the connectio first (i.e. the client)
+
+        m1.execute("ping -c 1 10.0.0.2")
```

Comment:
This is a race condition -- this won't work immediately after the `wg set wg0` on m2, supposedly it will take some time to establish the connection. The ping [failed here](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230704-110812-1ea465d9-fedora-38-other/log.html#191-1), presumably due to that. But that test failure is also rather unhelpful, there are zero error messages.

I recommend adding some debugging output if the ping fails, so that you can see the system state when it fails:
```py
try:
    m1.execute("ping -c 1 10.0.0.2")
except subprocess.CalledProcessError:
    m1.execute("ip a >&2; ip route >&2; nmcli c >&2")
    raise
```
and of course whatever else seems useful to you -- `nmcli d` perhaps, or there is something specific for wireguard, or even some `wg status`-like command.